aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -973,6 +973,81 @@ void xlog1py_kernel(TensorIteratorBase& iter) {
   });
 }
 
+void betainc_kernel(TensorIteratorBase& iter) {
+  AT_DISPATCH_FLOATING_TYPES_AND2(kBFloat16, kHalf, iter.common_dtype(), "betainc_cpu", [&]() {
+    cpu_kernel(iter, [](scalar_t x, scalar_t a, scalar_t b) -> scalar_t {
+      if (at::_isnan(x)){
+        return NAN;
+      }
+      if (at::_isnan(a)){
+        return NAN;
+      }
+      if (at::_isnan(b)){
+        return NAN;
+      }
+      if (x < 0.0 || x > 1.0 || a < 0.0 || b < 0.0){
+        return NAN;
+      }
+
+      /*The continued fraction converges nicely for x < (a+1)/(a+b+2)*/
+      /*Use the fact that beta is symmetrical.*/
+      bool return_inverse = false;
+      if (x > (a + 1.0) / (a + b + 2.0)) {
+        std::swap(a, b);
+        x = 1.0 - x;
+        return_inverse = true;
+      }
+
+      /*Find the first part before the continued fraction.*/
+      const scalar_t lbeta_ab = std::lgamma(a) + std::lgamma(b) - std::lgamma(a + b);
+      const scalar_t front = std::exp(std::log(x) * a + std::log(1.0 - x) * b - lbeta_ab) / a;
+
+      /*Use Lentz's algorithm to evaluate the continued fraction.*/
+      scalar_t f = 1.0, c = 1.0, d = 0.0;
+
+      scalar_t TINY = 1.0e-30, STOP = 1.0e-8, numerator, cd;
```

Comment:
Can TINY and STOP be constexpr

Better to default initialize `numerator` and `cd`

Reply:
If I put them to `constexpr` it gives an error but to `const` it works.